addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_tree.py

```diff
@@ -461,7 +461,7 @@ def recursive_filter_tag(self, uuid, parent_keep_tag):
         for child in self.nodes[uuid].children:
             if self.nodes[uuid].blender_type == VExportNode.INST_COLLECTION or self.nodes[uuid].is_instancier == VExportNode.INSTANCIER:
                 # We need to split children into 2 categories: real children, and objects inside the collection
-                if self.nodes[uuid].children_type[child] == VExportNode.CHILDREN_IS_IN_COLLECTION:
+                if self.nodes[uuid].children_type.get(child, VExportNode.CHILDREN_REAL) == VExportNode.CHILDREN_IS_IN_COLLECTION:
```

Comment:
Well, I think you should not try to "patch" this check, but a better solution is to find and fix the root cause of why the children_type is not set in that case